src/features/Common/data/slice.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+import { RequestStatus } from 'features/constants';
+
+const initialState = {
+  allClasses: {
+    data: [],
+    status: RequestStatus.INITIAL,
+    error: null,
+  },
+};
+
+export const commonSlice = createSlice({
+  name: 'common',
+  initialState,
+  reducers: {
+    fetchAllClassesDataRequest: (state) => {
+      state.allClasses.status = RequestStatus.LOADING;
+    },
+    fetchAllClassesDataSuccess: (state, { payload }) => {
+      state.allClasses.data = payload;
+      state.allClasses.count = payload?.length;
+      state.allClasses.status = RequestStatus.SUCCESS;
+    },
+    fetchAllClassesDataFailed: (state) => {
+      state.allClasses.status = RequestStatus.ERROR;
+    },
+  },
```

Comment:
Not sure about this, `fetchAllClassesDataRequest` the name suggest we're updating the state with classes but the reducer only update the **status** of the request, same for `fetchAllClassesDataFailed` which suggest we update the state with failed classes ( which does not have to much sense to have failed classes ), now what if I must handle a different `status` now I need "reset", I must add a new reducer for that? Why just don't have a general purpose reducer?

```js
    updateRequestClassStatus: (state, { payload }) => {
      state.allClasses.status = payload;
    },
```

Now we got something more flexible and clear without add more reducers, more easy to understand more easy to follow.


Now regarding with `fetchAllClassesDataSuccess` still thinking  the name is not clear, we aren't fetching anything, this is a reducer not a thunk so a better name could be `updateClasses` the "all" that's left in my opinion. But what do u think?


```suggestion
     updateRequestClassStatus: (state) => {
      state.allClasses.status = RequestStatus.LOADING;
    },
    updateClasses: (state, { payload }) => {
      state.allClasses.data = payload;
      state.allClasses.count = payload?.length;
      state.allClasses.status = RequestStatus.SUCCESS;
    },
  },
```